### PR TITLE
Milestone 2: User interface

### PR DESCRIPTION
**Pull request by realisticattorney**
Milestones for this step were:

- To create a visual representation of the gameboard

- To require input for the user moves in the terminal

### CODE REVIEW

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello @realisticattorney and @gasb150, Nice work 👏  

You've done nicely laying this all out. Please consider that you are missing the loop supposed to drive the code of the game. Consider making this addition and request your next review.

It's just to simulate how the game will run, even though you are already doing that 👍 , but use a loop, for example:

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```
In your basic flow, consider 
- game telling which player turn it is 👍 
- game asking the player to select from available moves
- game informing the player if the selected move is invalid (without computing what an invalid move is)
- game displaying board after the player move (without computing which spots on the board are taken) 👍 
- game informing the player if the selected move is a winning move (without computing the winning move)
- game informing the player if the selected move is a draw move (without computing draw)
- game repeats all actions for the next player's move 

I know am asking too much 😄, but you already have some `puts` statements, just add them and wrap around the game loop, then let it run once or two, that should be enough.

- [ ] By now you should kindly consider updating your readme file to reflect progress.

This won't take your time, do this and you'll be good to go.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    #        next
+    #      elsif chosen_p2 == rows_cols[i]
+    #        p 'PLAYER 2 WINS'
+    #        win = true
+    #        next
+    #      end
+    #     end
+    #  end
+
+  end
```
Comment (peterrobert) [inline bin/main.rb]: Nice work adding a while loop to your project just like suggested above. However, the way you have approached it is a bit wrong kindly consider the following :arrow_down: 

- [ ] You should not use a number to terminate your loop. For instance, you are using 6 as the number of times your loop should run which automatically tells me something is wrong. A loop needs to be terminated at some point (after a winning move or draw to be more specific). Kindly consider using a variable. 

Example: 
`
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
````

**Nb:** Remember you don't need to compute a draw move or a winning move

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
HELLO @realisticattorney  :wave: NICE PROGRESS WITH THIS MILESTONE. :+1: I LOVE THAT YOU HAVE IMPLEMENTED SOME CHANGES JUST LIKE REQUESTED BY THE PREVIOUS TSE. HOWEVER, THERE ARE A FEW MORE CHANGES TO BE DONE KINDLY CHECK BELOW :arrow_down: 

- [ ] `[REQUIRED]` game informs the player if the selected move is invalid (without computing what an invalid move is). I can see that the game is detecting available moves. Kindly also consider notifying the user that the move is invalid. You can do so by outputting a message.
 
### [OPTIONAL] 

- [ ] Kindly consider asking the players for their names.

### kindly make those changes and submit your project for another review

### if you have any question kindly reach me on slack @peterrobert

### HAPPY CODING :computer:

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Tavo Sanmartin Beleño & German Aquila** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] Remove commented code in main.rb
- [x] Main should have "puts" or "print" to show the winner if there is one
- [ ] Include the rubocop file in your project
- [ ] Include linter configuration for your project

Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

## Hello team 😃,

**Great job implementing the requested changes 👏. However, there are still a few fixes needed.**

1). **The linter checks are not displaying on your pull request.**
linters

2). **Correct the name of your `rubocop` file. It should be `.rubocop.yml`.** 

## Happy coding 🥂 

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats team, you guys did an excellent job, you have my approval to go ahead and merge this pull request!


⚠ Kindly note that you should update your readme to show your progress ⚠

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**